### PR TITLE
Suggested changes to the readme

### PR DESCRIPTION
**Pull request by Cleop**
I've gone through the readme and picked out any typos that I could see, added an extra link for another learning resource etc. I think my linter on atom has changed some of the formatting of the file so it looks like everything's changed when it hasn't 🙈! 

The only practical change is that around step 3 there was a typo on the name of the `hash_password` fn which in places was called `hash_password` but somewhere else was just called `hash` and this meant my tests weren't passing. Having now corrected this people should not encounter this error. The same also applies at step 8 where:

```
  def dump(value) do
    {:ok, hash(value)}
  end
```
is now `hash_password(value)`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,49 +47,50 @@ We are _following_ a **_battle-tested_ industry-standard** approach
...
-as a "Git Submodule".
+* **Advanced Encryption Standard** (**AES**) to encrypt sensitive data.
+  * **Galois/Counter Mode**
+    for _symmetric_ key cryptographic block ciphers:
+    https://en.wikipedia.org/wiki/Galois/Counter_Mode
+    recommended by many security and cryptography practitioners including
+    [Matthew Green](https://blog.cryptographyengineering.com/),
+    [Niels Ferguson](https://en.wikipedia.org/wiki/Niels_Ferguson)
+    and [Bruce Schneier](https://www.schneier.com/blog/about/)
+  * "Under the hood" we are using Erlang's
```
Comment (nelsonic) [inline README.md]: I _prefer_ to use the `+` for bullet-points because the `*` is often used fro _emphasis_ ...

File: README.md [hunk 2]
```diff
@@ -47,49 +47,50 @@ We are _following_ a **_battle-tested_ industry-standard** approach
...
 Cryptographic and Password rules:
-+ [x] Use "***strong approved Authenticated Encryption***"
-based on an ***AES algorithm***.
-  + [x] Use GCM mode of operation for symmetric key cryptographic block ciphers.
-  + [x] Keys used for encryption must be rotated at least annually.
-+ [x] Only use approved public algorithm **SHA-256** or better for hashing.
-+ [x] **Argon2** is the winner of the password hashing competition
-and should be your ***first choice*** for **_new_ applications**.
+
+* [x] Use "**_strong approved Authenticated Encryption_**"
```
Comment (nelsonic) [inline README.md]: @Cleop which Markdown formatter are you using that automatically changes `+` (_bullet points_) to `*`...?

File: README.md [hunk 3]
```diff
@@ -157,19 +156,19 @@ will take around **1 hour** <br />
...
+> and [**fines**](https://www.itgovernance.co.uk/dpa-and-gdpr-penalties)
+> of a data breach_.
 
 ## How?
 
 These are "step-by-step" instructions,
 don't skip any step(s).
 
-### 1. Creat the `encryption` App
+### 1. Create the `encryption` App
```
Comment (nelsonic) [inline README.md]: 👍

File: README.md [hunk 4]
```diff
@@ -687,31 +683,28 @@ in case the DB is ever "compromised"
...
+where we want a deterministic digest_).
 
 Using `argon2` makes "cracking" a password
 (_in the event of the database being "compromised"
 far less likely_) as is has _both_ a CPU-bound "work-factor"
 _and_ a "Memory-hard" algorithm which will _significantly_
 "slow down" the attacker.
 
-##### Add the `argon2` Dependency to
+##### Add the `argon2` Dependency
```
Comment (nelsonic) [inline README.md]: this should _actually_ be:
Add the `argon2` Dependency to `mix.exs` 
but the suggested change is good too. (_more concise_) 👍

File: README.md [hunk 5]
```diff
@@ -731,26 +723,27 @@ The first function we need is `hash_password/1`:
 ```elixir
 defmodule Encryption.PasswordField do
 
-  def hash(value) do
+  def hash_password(value) do
```
Comment (nelsonic) [inline README.md]: good catch. thanks for fixing.

File: README.md (hunk 2 above)
Comment (Cleop) [reply README.md]: Nice points re plugins, I agree. I just took a look at my packages and I have a grand total of 5 linters installed! My guess would be that this is from 'prettier-atom' which is the one I remember adding most recently for one of our projects. It changes a lot more than any other linters I've used before I think. I think picking some as a team would be great 👍

**Comment by Cleop** [comment]
@nelsonic - Would you like me to pick out the linter changes if I skim through the commit so we can avoid these kind of changes?

**Comment by nelsonic** [comment]
@Cleop we _either_ need to _standardise_ the Markdown format by _everyone_ using the same Plugin
***or*** avoid wholesale changes by _disabling_ the plugin ...

> I would not "_accept_" a PR form a _community_ member that made this many formatting changes.
> The "signal-to-noise" ratio is too high.

File: README.md (hunk 2 above)
Comment (nelsonic) [reply README.md]: please open an issue on Linters in: https://github.com/dwyl/technology-stack/issues
so we can _capture_ the discussion centrally. thanks.

File: README.md [hunk 6]
```diff
@@ -687,31 +683,28 @@ in case the DB is ever "compromised"
...
-[`test/lib/hash_field_test.exs`](https://github.com/dwyl/phoenix-ecto-encryption-example/blob/master/test/lib/hash_field_test.exs)
-
-
+> [`test/lib/hash_field_test.exs`](https://github.com/dwyl/phoenix-ecto-encryption-example/blob/master/test/lib/hash_field_test.exs)
 
 #### 3.5 Hash _Password_
 
 When hashing **passwords**, we want to use the **_strongest_ hashing algorithm**
-and we also want the hashed value (_or "digest"_) to be ***different****
+and we also want the hashed value (_or "digest"_) to be **\*different\*\***
```
Comment (nelsonic) [inline README.md]: I suspect that the change here is `***different****` to `***different****` (_the typo is an extra asterisk after he word_) no need for any character escaping (_back slashes_)

**Comment by Cleop** [comment]
Closing in favour of #14

**Comment by nelsonic** [comment]
Thanks @Cleop 🥇

**Pull request closed without merging**